src/main/java/uk/gov/justice/services/test/domain/EventsChecker.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.justice.services.test.domain;
+
+import static uk.gov.justice.services.test.DomainTest.eventNameFrom;
+import static uk.gov.justice.services.test.DomainTest.generatedEventAsJsonNode;
+import static uk.gov.justice.services.test.DomainTest.jsonNodeWithoutMetadataFrom;
+
+import uk.gov.justice.services.test.exception.EventException;
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.commons.lang3.StringUtils;
+
+public class EventsChecker {
+    public static void compareEvents(List<JsonNode> expectedEvents, List<Object> generatedEvents) {
+        if (expectedEvents.size() > generatedEvents.size()) {
+            throw new EventException(createMessage(expectedEvents, generatedEvents, null));
```

Comment:
Is throwing an unchecked exception really the best model for this? How does this message look when the exception gets thrown when the test steps are run? We don't really want a stack trace in the output, we want an output like a test assertion failing, like the original assertThat calls.